index.js

```diff
@@ -1,121 +1,74 @@
-        .then(users => {
-            if (typeof users.rows === 'object' && users.rows.length > 0){
-                // first user is always admin user, let's not mess with them
-                users.rows.shift();
-                console.log("Found",users.rows.length,"total users.");
-                return users.rows;
-            } else {
-                console.log('No users found');
-                process.exit(1);
-            }
-        })
-        .catch(e => {
-            handleError(e);
-        })
+const getAllUsers = async () => {
+    const uri = compileUrl('/_users/_all_docs');
+    return request.get({ uri, json: true});
 }
 
-const getUserPerms = function (allUsers){
-    allUsers.forEach((username)=>{
-        const userId = username.id;
-        const usernameAry = userId.split(':');
-        try {
-            compiledUrl = new URL('/_users/' + userId, couch_url);
-        } catch(e) {
-            console.log('Error while creating url: ', e.message);
-            process.exit(1);
-        }
+const filterOutAdmins = (users) => {
+    if (typeof users.rows === 'object' && users.rows.length > 1){
+        // first user is always admin user, let's not mess with them
+        users.rows.shift();
+        console.log(`Found ${users.rows.length} total users.`);
+        return users.rows;
+    }
+    return [];
+};
 
-        const options = {
-            uri: compiledUrl.href,
-            json: true
-        };
+const getUpdatedUsers = async (allUsers) => {
+    const updates = [];
+    for (const user of allUsers) {
+        const uri = compileUrl('/_users/' + user.id);
+        const user = await request.get({ uri, json: true });
+        if (typeof user.roles === 'object' && !user.roles.includes(role_to_use)) {
+            user.roles.push(role_to_use);
+            updates.push(users);
+        }
+    }
+    return updates;
+}
 
-        return request.get(options)
-            .then(userData => {
-                if (typeof userData.roles === 'object'){
-                    if (!userData.roles.includes(role_to_use)){
-                        userData.roles.push(role_to_use);
-                        setPermission(userId, {
-                                'roles': userData.roles,
-                                '_rev': userData._rev,
-                                'type': "user",
-                                'name': usernameAry[1],
-                            }
-                        );
-                    }
-                }
-            })
-            .catch(e => {
-                handleError(e);
-            });
-    })
+const saveUsers = async (users) => {
+    for (const user of users) {
+        const userId = user._id.split(':')[1];
+        const uri = compileUrl('/_users/' + userId);
+        await request.put({ uri, json: true, body: user });
+    }
 }
 
-const setPermission = function(userId, userRoles){
-    let compiledUrl
-    let options;
+const go = async () => {
```

Comment:
This is a bit of a hack because at least some versions of Node don't allow async at the top level. I think this works now, but I haven't actually executed it!

Reply:
i'll merge and test!